### ISSUE TITLE
Run E2E tests in parallel

### ISSUE THREAD

**Issue by ezodude** [kubeshop/botkube#749]
To speed up CI builds and pull request builds/reviews we should run the E2E tests in parallel.

To solve this, [two options were suggested](https://github.com/kubeshop/botkube/issues/307#issuecomment-1249681549).

IMO, the simpler and easier to maintain option is: Do away with helm test and run `make test-integration-(integration)` in parallel.

This can be implemented as follows,

- Ensure all required ENV VARS are loaded, including `SLACK_BOT_NAME`.

- Use the [GitHub Actions matrix strategy](https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#jobsjob_idstrategymatrix) to run `make test-integ...` jobs in parallel.

In practice this means updating `pr-build.yaml` and `branch-build.yaml` with:

```YAML
  ...
  integration-tests:
  ...
  strategy:
      matrix:
        integration:
          - slack
          - discord
  steps:
  ...
    - name: Run tests
      run: "make test-integration-${{ matrix['integration'] }}"
  ...
```

**Comment by pkosiec** [comment]
Few questions:

1. is it already agreed to have in the 0.14 scope? I would recommend postponing it to 0.15 as we still have quite a lot of work to do.

> IMO, the simpler and easier to maintain option is: Do away with helm test and run `make test-integration-(integration)` in parallel.
>
> This can be implemented as follows,
>
> - Ensure all required ENV VARS are loaded, including `SLACK_BOT_NAME`.
>
> - Use the [GitHub Actions matrix strategy](https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#jobsjob_idstrategymatrix) to run `make test-integ...` jobs in parallel.

2. So that would mean the setup (randomizing channel names, updating deployment with envs) would be done before running actual tests using matrix? How would we like to do implement this, some simple Go code ran in pipeline?

3. Did we consider keeping what we have and using `t.Parallel()a and moving the initial setup + cleanup to `TestMain`? 🤔  That would work as well, right, or do you see any blockers for such solution?

**Comment by mszostok** [comment]
After discovering that we won't be able to add test with "restarts", IMO the solution suggested by Ezo would work nice. But we should not only run the tests but basically create the whole environments (create k3d cluster, etc.), so they will be properly isolated and parallelized.

**Comment by ezodude** [comment]
I am very happy to postpone this.

> 2. So that would mean the setup (randomizing channel names, updating deployment with envs) would be done before running actual tests using matrix? How would we like to do implement this, some simple Go code ran in pipeline?
> 3. Did we consider keeping what we have and using t.Parallel() and moving the initial setup + cleanup to TestMain? 🤔 That would work as well, right, or do you see any blockers for such solution?

But to clarify, the idea is to go for a very simple solution that avoids the above.

@mszostok's would make using the strategy matrix even easier.

**... Again ... happy to de-scope for v0.14.0**

**Comment by pkosiec** [comment]
>But to clarify, the idea is to go for a very simple solution that avoids the above.

Ah, okay, so we'll run the Deployment with properly set envs already as a separate K8s cluster (in matrix). No need to restart it and set/restore envs. Makes sense 👍 

> ... Again ... happy to de-scope for v0.14.0

Just wanted to point out that we probably won't manage to do it - of course if anyone from the team will have capacity to deliver it for 0.14, then that would be awesome 🚀 Thanks!
